leezer3/OpenBVE
# New: Running on other Rails (WIP)

This is a followup to #326 , and introduces the ability for the player's train to run on other rails within the world.

Heavily WIP.

### Current Usage:
**Track.Switch firstRail; secondRail; initialSetting; switchName; firstTrackName; secondTrackName**
Creates a standard facing switch.
* FirstRail : The rail the switch is placed on.
* secondRail : The rail that may be switched to.
* initialSetting : The initial setting for the switch.
* switchName: The display name for the switch in the menu etc. (e.g. Yard Entry Switch)
* firstTrackName: The display name for the first track (e.g. Mainline)
* secondTrackName: The display name for the second track (e.g. Yard)

**Track.SwitchT firstRail; secondRail; initialSetting; switchName; firstTrackName; secondTrackName**
Creates a trailing switch.
* FirstRail : The rail the switch is placed on.
* secondRail : The rail that is joined to.
* initialSetting : The initial setting for the switch. *Note:* To set the trailing switch to start in a position where we can run through it, this must be set to the rail which it is placed upon.
* switchName: The display name for the switch in the menu etc. (e.g. Yard Entry Switch)
* firstTrackName: The display name for the first track (e.g. Mainline)
* secondTrackName: The display name for the second track (e.g. Yard)

### Flaws:
N.B. These will change, just need the correct code implementing & thinking about.
* ~~Only works for branching switches.~~
* ~~No direction detection.~~
* ~~Running past a .RailEnd command is likely to crash.~~
* 3 way & greater switches not supported by the current command structure. Need to think about how we could structure this.....
* ~~Needs a way to change switches. (Leverage #317 to highlight the switch?)~~
* ~~Needs sprung return points adding (simples....)~~
* ~~Point sound events are not generated at the minute on other tracks.~~

### To Test:
* Check all combinations.
* Check for conflicts with #326
